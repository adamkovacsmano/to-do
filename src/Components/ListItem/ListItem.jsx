import React, { Component } from "react";
import styles from "./ListItem.module.scss";

class ListItem extends Component {
  render() {
    const markedStyle = this.props.isMarked ? styles.marked : "";
    return (
      <div className={styles.listContainer}>
        <li
          type="text"
          className={markedStyle}
          onClick={() => this.props.markAsDone(this.props.data.docId)}
        >
          {this.props.text}
        </li>
        <span className={styles.userInfo}>
          {" "}
          ...by {this.props.userName}{" "}
          <img
            src={this.props.userImage}
            alt="user img"
            width="20px"
            height="20px"
          ></img>
        </span>
        <button
          id="testButton3"
          className={styles.delete}
          onClick={() => this.props.deleteItem(this.props.data.docId)}
        >
          X
        </button>
        <button
          className={styles.edit}
          onClick={() => this.props.updateItem(this.props.data.docId)}
        >
          edit
        </button>
      </div>
    );
  }
}

export default ListItem;
