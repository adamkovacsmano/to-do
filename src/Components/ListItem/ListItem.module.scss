.listContainer {
  display: flex;
  li {
    cursor: pointer;
  }
  .userInfo {
    margin-left: 5px;
    font-style: italic;
  }
}

.marked {
  text-decoration: line-through;
}

.delete {
  background-color: transparent;
  border: none;
  font-weight: bolder;
  cursor: pointer;
  margin-left: 10px;
}

.edit {
  background-color: transparent;
  border: none;
  cursor: pointer;
  margin-left: 10px;
  font-style: italic;
}
