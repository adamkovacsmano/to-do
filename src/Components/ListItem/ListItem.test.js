import React from "react";
import ListItem from "./ListItem";
import { shallow, mount } from "enzyme";

describe("FilterIcon Tests", () => {
  let listItem;
  beforeEach(() => {
    listItem = shallow(<ListItem />);
  });

  test("should match the snapshot", () => {
    expect(listItem).toMatchSnapshot();
  });
});

describe("ListItem Tests", () => {
  let event;

  let data = {
    text: "lorem",
    isMarked: false,
    userName: "john dove",
    userImage:
      "https://lh3.googleusercontent.com/a-/AAuE7mBtNtlAK6pzmOGy2AxSBDaUeGB56EhdQlp8AXSGCg"
  };

  beforeEach(() => {
    event = shallow(<ListItem eventData={data} />);
  });

  test("should match the snapshot", () => {
    expect(event).toMatchSnapshot();
  });
});

//-------------------------------------- docId undefined for both tests:

// const clickFn = jest.fn();
// describe("button", () => {
//   it("button click should delete", () => {
//     const component = shallow(<ListItem onClick={clickFn} />);
//     component.find("button#testButton3").simulate("click");
//     expect(clickFn).toHaveBeenCalled();
//   });
// });

// const mockFunction = jest.fn();
// it("should call mockFunction on button click", () => {
//   const component = mount(<ListItem onClickFunction={mockFunction} />);
//   component.find("button#testButton3").simulate("click");
//   expect(mockFunction).toHaveBeenCalled();

//   component.unmount();
// });
