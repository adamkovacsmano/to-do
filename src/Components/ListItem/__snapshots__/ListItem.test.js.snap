// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterIcon Tests should match the snapshot 1`] = `
<div
  className="listContainer"
>
  <li
    className=""
    onClick={[Function]}
    type="text"
  />
  <span
    className="userInfo"
  >
     
    ...by 
     
    <img
      alt="user img"
      height="20px"
      width="20px"
    />
  </span>
  <button
    className="delete"
    id="testButton3"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="edit"
    onClick={[Function]}
  >
    edit
  </button>
</div>
`;

exports[`ListItem Tests should match the snapshot 1`] = `
<div
  className="listContainer"
>
  <li
    className=""
    onClick={[Function]}
    type="text"
  />
  <span
    className="userInfo"
  >
     
    ...by 
     
    <img
      alt="user img"
      height="20px"
      width="20px"
    />
  </span>
  <button
    className="delete"
    id="testButton3"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="edit"
    onClick={[Function]}
  >
    edit
  </button>
</div>
`;
